feature-libs/asm/package.json

```diff
@@ -29,12 +29,15 @@
     "@angular/common": "^15.2.4",
     "@angular/core": "^15.2.4",
     "@angular/forms": "^15.2.4",
+    "@angular/platform-browser": "^15.2.4",
     "@ng-select/ng-select": "^10.0.3",
     "@ngrx/effects": "^15.3.0",
     "@ngrx/store": "^15.3.0",
     "@spartacus/cart": "6.3.0-1",
     "@spartacus/core": "6.3.0-1",
+    "@spartacus/order": "6.3.0-1",
     "@spartacus/schematics": "6.3.0-1",
+    "@spartacus/storefinder": "6.3.0-1",
```

Comment:
[MAJOR REMARK - breaking change]
Adding new required peerDependencies of a package `@spartacus/asm` is a breaking change. It will cause compilation errors for customers who used to use `@spartacus/asm` library, but didn't have installed e.g. `@spartacus/storefinder`.

When they upgrade to 6.5, they'll get compilation errors because the npm package `@spartacus/storefinder` is not installed. The fix for them would be to install this library manually (`npm install @spartacus/storefinder`).

IMHO this _unavoidable_ breaking change means to me that that this feature should be released only in a Major version of Spartacus.

Alternatively we might want to accept it (with PO's agreement) that we turn a blind eye on that breaking change, as long as it's well communicated to customers.

Reply:
UPDATE:
We've done an ultimate test, whether there will be any compilation errors or npm warnings when updating from 6.5 to libraries from this PR. And there were no errors nor warnings.

The test was made this way:
1. create a fresh Angular 15 app - `npx @angular/cli@15 new test-app`
1. install Spartacus 6.5 with ASM (e.g. using RBSC as npm repository) - `ng add @spartacus/schematics), BUT without Storefinder(!) - uncheck the `Store Finder` feature checkbox in the installation schematics prompt
1. verify in `package.json` of the fresh app there is no `@spartacus/storefinder` mentioned
1. verify in `node_modules` of the fresh app there is no `node_modules/@spartacus/storefinder` folder

1. then build the libraries on our local epic branch and publish them to local verdaccio npm repo (for more see, [docs](https://github.com/SAP/spartacus/blob/0dd42d4ab0121265024ee092d7cb3ef8640bf475/projects/schematics/README.md#L16) and our [script ](https://github.com/SAP/spartacus/blob/0dd42d4ab0121265024ee092d7cb3ef8640bf475/tools/schematics/testing.ts#L215) for publishing to local verdaccio repository) 
1. then in the fresh app bump the version of spartacus:
   1. change `~6.5.0-1` and `^6.5.0-1` versions of `@partacus/...` packages to _exaclty_ `6.5.0-1` (with no tilde or carret!)
   3. run `npm update`
1. Verify there were no warnings during the `npm update`
2. Verify the `package.json` of the fresh app NOW contains `@spartacus/storefinder` entry (it was installed implicitly)
3. (Refresh the files tree view in your IDE and) verify the folder `node_modules/@spartacus/storefinder` exists

Reply:
For reference:
**Reason why there were no npm warnings:**
During the `npm update`, NPM discoveres that the existing primary dependency `@spartacus/asm` has a new peerDependency (`@spartacus/storefinder`) and installs it automatically behind the scenes. That said, NPM doesn't put `@spartacus/storefinder` explicitly in `package.json` as a primary dependency, but you can find notion about `@spartacus/storefinder` only in the updated `package-lock.json` and also in the newly-added folder `node_modules/@spartacus/storefront`.

**Reason why there were no compilation errors:**
Since the `@spartacus/storefront` library is installed impliclty by `npm update`, then there is no potential reason for compilation error due to lack of `@spartacus/storefront`